pythoncz_tests/models/github_test.py

```diff
@@ -0,0 +1,432 @@
+                                issues=[issue_org1], pull_requests=[]),
+        ]),
+    )
+    issue_org2 = fixtures.issue()
+    requests_mock.add(
+        responses.POST,
+        'https://api.github.com/graphql',
+        json=fixtures.api_response_body(repos=[
+            fixtures.repository(private=False,
+                                issues=[issue_org2], pull_requests=[]),
+        ]),
+    )
+
+    issues = github.get_issues(['org1', 'org2'])
+    titles = set(i['title'] for i in issues)
+
+    assert titles == {issue_org1['title'], issue_org2['title']}
+
+
+def test_get_issues_api_error(github, requests_mock):
+    """
+    In case of API error, the 'get_issues' function should raise HTTPError
+    with combined error messages
+    """
+    requests_mock.add(
+        responses.POST,
+        'https://api.github.com/graphql',
+        status=500,
+        json={'errors': [{'message': 'Error 1'}, {'message': 'Error 2'}]},
+    )
+    error_message = 'Error 1; Error 2'
+    with pytest.raises(requests.HTTPError, match=error_message) as excinfo:
+        github.get_issues(['org'])
+
+    assert excinfo.value.response.status_code == 500
+
+
+def test_create_api_session(github):
+    """
+    The '_create_api_session' helper should add User-Agent and Authorization
+    HTTP headers to the HTTP session
+    """
+    session = github._create_api_session()
+    headers = session.headers
+
+    assert 'https://python.cz' in headers.get('User-Agent')
+    assert re.match(r'token \w+', headers.get('Authorization')) is not None
+
+
+def test_get_issues_for_org_key_error(github, requests_mock):
+    """
+    If the API response doesn't have the expected structure, the
+    '_get_issues_for_org' helper should raise ValueError
+    """
+    requests_mock.add(
+        responses.POST,
+        'https://api.github.com/graphql',
+        json={'data': {}},
+    )
+    error_message = 'Unexpected structure of the GitHub API response'
+    with pytest.raises(ValueError, message=error_message):
+        list(github._get_issues_for_org(requests.Session(), 'org'))
+
+
+def test_get_issues_for_org_merges_issues_pull_requests(github, requests_mock):
+    """
+    The '_get_issues_for_org' helper should merge both issues with
+    pull requests and treat them both as issues
+    """
+    repo1 = fixtures.repository(
+        issues=[fixtures.issue(), fixtures.issue(), fixtures.issue()],
+        pull_requests=[fixtures.issue(pr=True)],
+        private=False
+    )
+    repo2 = fixtures.repository(
+        issues=[fixtures.issue(), fixtures.issue()],
+        pull_requests=[fixtures.issue(pr=True), fixtures.issue(pr=True)],
+        private=False
+    )
+
+    expected_titles_are_pr = {}
+    for repo in [repo1, repo2]:
+        for issue in repo['issues']['nodes']:
+            expected_titles_are_pr[issue['title']] = False
+        for pull_request in repo['pullRequests']['nodes']:
+            expected_titles_are_pr[pull_request['title']] = True
```

Comment:
I don't mind it's new, it just felt to me this particular case isn't much more readable with the comprehensions.

But ❗️ the production version is 3.4 so this won't work on production 😱 @hroncok And we accepted the other comprehension.